datafusion/sql/src/unparser/utils.rs

```diff
@@ -127,3 +159,30 @@ pub(crate) fn unproject_window_exprs(expr: &Expr, windows: &[&Window]) -> Result
         })
         .map(|e| e.data)
 }
+
+fn find_agg_expr<'a>(agg: &'a Aggregate, column: &Column) -> Result<Option<&'a Expr>> {
+    if let Ok(index) = agg.schema.index_of_column(column) {
+        if matches!(agg.group_expr.as_slice(), [Expr::GroupingSet(_)]) {
+            // For grouping set expr, we must operate by expression list from the grouping set
+            let grouping_expr = grouping_set_to_exprlist(agg.group_expr.as_slice())?;
```

Comment:
**Concern:**

`grouping_set_to_exprlist` is used directly in `find_agg_exp` so invoked for each column to unproject (in case of grouping set). This could be optimized to do this only once. Decided to keep this logic this way as pre-calculating and passing such expressions as argument or changing `find_agg_exp` signature will make code less clean and `grouping_set_to_exprlist` does NOT create expressions copies (operates by Vec<&Expr>)